### ISSUE TITLE
`blocTest` `errors` parameter only catches subclasses of `Exception`

### ISSUE THREAD

**Issue by cameron1024** [felangel/bloc#2466]
**Describe the bug**
The `errors` parameter of `blocTest` only works with types that extend/implement `Exception` (`flutter_bloc: 8.0.0`)

For context, I have a bloc that has a method that should only be called when it is in a certain state. This could be an `assert`, but I'd like to keep behaviour similar between test/prod, so I use `throw AssertionError()` instead. I'd also like to verify this behaviour in my tests, because I rely on this error being thrown and being visible in application logs.

Having a look at the code for `blocTest`, it looks like this was perhaps intentional though (maybe this issue should be a feature request instead, sometimes it's hard to tell), so if there's a more idiomatic way of doing this, that would be great too.

I'm sure I could achieve this manually by using `testBloc` and `runZoned` and suppressing the `visibleForTesting` error, but I'd rather avoid doing that if at all possible.

**To Reproduce**
```
class TestBloc extends Cubit<int> {
  TestBloc() : super(0);
  void test() {
    throw Error();
  }
}

void main() {
  blocTest<TestBloc, int>(
    'test',
    build: () => TestBloc(),
    act: (bloc) => bloc.test(),
    errors: () => [isA<Error>()],
  );

}
```
This currently fails, the test throws an uncaught error. If I replace `Error` with `Exception` the test passes. Using any other non-exception type (e.g. `String`) also fails

**Expected behavior**
This test should catch the `Error` and the test should pass.

**Screenshots**
N/A
**Logs **
N/A

```
[√] Flutter (Channel unknown, 2.0.6-0.0.pre.1, on Microsoft Windows [Version 10.0.19042.985], locale en-GB)
    • Flutter version 2.0.6-0.0.pre.1 at C:\Users\cameron\flutter
    • Framework revision 16e8e53e67 (3 weeks ago), 2021-05-01 00:19:01 +0100
    • Engine revision b09f014e96
    • Dart version 2.12.3

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\cameron\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.56.2)
    • VS Code at C:\Users\cameron\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.22.0

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 90.0.4430.212
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 90.0.818.51

• No issues found!
```

**Additional context**

**Comment by cameron1024** [comment]
I've created a PR that fixes this: #2469 , let me know if it needs anything else before it can be merged :-)

**Comment by felangel** [comment]
Closing this since it is addressed in #2469 and will be included in the next release of bloc_test (very shortly)
